napari/layers/intensity_mixin.py

```diff
@@ -124,9 +126,14 @@ def contrast_limits_range(self, value):
         # and updating the views/controllers
         if hasattr(self, '_contrast_limits') and any(self._contrast_limits):
             cur_min, cur_max = self.contrast_limits
-            new_min = min(max(value[0], cur_min), value[1])
-            new_max = max(min(value[1], cur_max), value[0])
-            self.contrast_limits = (new_min, new_max)
+            # if range limits are outside of current limits, override
+            if value[0] > cur_max or value[1] < cur_min:
+                self.contrast_limits = tuple(value)
```

Comment:
This is a side-effect of a publicly settable property of some important layers, so this behavior should be documented in the docstring. Some of the side effects were not previously documented, but it would be good to improve things here.

I considered raising here instead, but that seems unnecessarily picky. I also considered warning, but if a user is setting this particular property I think they should know what they're doing.